livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java

```diff
@@ -0,0 +1,364 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+
+/***
+ * Livy helper class
+ */
+public class LivyHelper {
+  Logger LOGGER = LoggerFactory.getLogger(LivyHelper.class);
+  Gson gson = new GsonBuilder().setPrettyPrinting().create();
+  HashMap<String, Object> paragraphHttpMap = new HashMap<>();
+  Properties property;
+
+  LivyHelper(Properties property) {
+    this.property = property;
+  }
+
+  public Integer createSession(InterpreterContext context, String kind) throws Exception {
+    try {
+      String json = executeHTTP(property.getProperty("zeppelin.livy.url") + "/sessions",
+          "POST",
+          "{\"kind\": \"" + kind + "\", \"proxyUser\": \""
+              + context.getAuthenticationInfo().getUser() + "\"}",
+          context.getParagraphId()
+      );
+      Map jsonMap = (Map<Object, Object>) gson.fromJson(json,
+          new TypeToken<Map<Object, Object>>() {
+          }.getType());
+      Integer sessionId = ((Double) jsonMap.get("id")).intValue();
+      if (!jsonMap.get("state").equals("idle")) {
+        while (true) {
+          LOGGER.error(String.format("sessionId:%s state is %s",
+              jsonMap.get("id"), jsonMap.get("state")));
+          Thread.sleep(1000);
+          json = executeHTTP(property.getProperty("zeppelin.livy.url") + "/sessions/" + sessionId,
+              "GET", null,
+              context.getParagraphId());
+          jsonMap = (Map<Object, Object>) gson.fromJson(json,
+              new TypeToken<Map<Object, Object>>() {
+              }.getType());
+          if (jsonMap.get("state").equals("idle")) {
+            break;
+          } else if (jsonMap.get("state").equals("error")) {
+            json = executeHTTP(property.getProperty("zeppelin.livy.url") + "/sessions/" +
+                    sessionId + "/log",
+                "GET", null,
+                context.getParagraphId());
+            jsonMap = (Map<Object, Object>) gson.fromJson(json,
+                new TypeToken<Map<Object, Object>>() {
+                }.getType());
+            String logs = StringUtils.join((ArrayList<String>) jsonMap.get("log"), '\n');
+            LOGGER.error(String.format("Cannot start  %s.\n%s", kind, logs));
+            throw new Exception(String.format("Cannot start  %s.\n%s", kind, logs));
+          }
+        }
+      }
+      return sessionId;
+    } catch (Exception e) {
+      LOGGER.error("Error getting session for user", e);
+      throw e;
+    }
+  }
+
+  protected void initializeSpark(final InterpreterContext context,
+                                 final Map<String, Integer> userSessionMap) throws Exception {
+    interpret("val sqlContext= new org.apache.spark.sql.SQLContext(sc)\n" +
+        "import sqlContext.implicits._", context, userSessionMap);
```

Comment:
I thought this 
`org.apache.spark.sql` or
`import sqlContext.implicits._`
only works on newer version of spark (post 1.3 I think) - should we just say only new versions are supported?